pkg/operators/common/kubeinventorycache.go

```diff
@@ -198,6 +241,45 @@ func (cache *inventoryCache) GetSvcs() ([]*v1.Service, error) {
 	return svcs, nil
 }
 
+func (cache *inventoryCache) GetSvcByName(namespace string, name string) *v1.Service {
+	cache.svcsMutex.Lock()
+	defer cache.svcsMutex.Unlock()
+	cache.removeOldSvcs()
+
+	key := namespace + "/" + name
+	if svc, ok := cache.svcs[key]; ok {
+		return svc
+	}
+	if oldSvc, ok := cache.oldSvcs[key]; ok {
+		return oldSvc.obj
+	}
+	return nil
+}
+
+func (cache *inventoryCache) GetSvcByIp(ip string) *v1.Service {
```

Comment:
Could you give me an example what you mean with similar approach to `Get()`?

In my head I would either need a type switch then (because of different stored objects) or I can create a general `Get(func selector(*T) bool)`. So create a Get function which receives a "selection function"

With the current Get(string key) its easy since this doesn't depend on `T`

Reply:
I mean if we go this route why not generalize:

```golang
func (cache *inventoryCache) GetCmp(cmp func(T, string) bool) T {
	cache.PruneOldObjects()
	cache.Lock()
	defer cache.Unlock()

	for _, value := range cache.current {
		if cmp(value, ip) {
			return value
		}
	}
	for _, value := range cache.old {
		if cmp(value, ip) {
			return value.obj
		}
	}
	return nil
}

func (cache *inventoryCache) GetPodByIp(ip string) *v1.Pod {
	return cache.GetCmp(func(a T, b string) bool { return a.Status.PodIP == ip})
}
```